test/test_scope_exit.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "rcpputils/scope_exit.hpp"
+
+TEST(test_scope_exit, should_run) {
+  bool called = false;
+
+  {
+    auto on_exit =
+      rcpputils::make_scope_exit(
+      [&called]() {
+        called = true;
+      });
+    (void)on_exit;
+  }
+
+  EXPECT_TRUE(called);
+}
+
+TEST(test_scope_exit, should_not_run) {
+  bool called = false;
+
+  {
+    auto on_exit =
+      rcpputils::make_scope_exit(
+      [&called]() {
+        called = true;
+      });
+    on_exit.cancel();
+  }
+
+  EXPECT_FALSE(called);
+}
+
+TEST(test_scope_exit, code_types) {
+  auto code = []() {};
+  rcpputils::make_scope_exit(code);
+
+  const auto const_code = []() {};
+  rcpputils::make_scope_exit(const_code);
+
+  struct noncopyable_code
+  {
+    noncopyable_code() = default;
+    noncopyable_code(const noncopyable_code &) = delete;
+    noncopyable_code(noncopyable_code &&) = default;
+    void operator()() {}
+  };
+  rcpputils::make_scope_exit(noncopyable_code());
```

Comment:
```suggestion
  rcpputils::make_scope_exit(noncopyable_code);
```
??

Reply:
Hmm, `noncopyable_code` is a type. Maybe a bit misleading.

Reply:
Sorry, my bad (I thought you already instantiated the object, and were calling the `operator()`).
I would use CamelCase per the style guide.

Reply:
See https://github.com/ros2/rcpputils/pull/78/commits/1360bc9071bda3eda708c31e91e029bf3b8dbfe5.

Reply:
From https://index.ros.org/doc/ros2/Contributing/Code-Style-Language-Versions/#function-and-method-naming:

> Note that classes should still use CamelCase by default